darwin/dataset/remote_dataset.py

```diff
@@ -333,9 +333,15 @@ def pull(
                 max_workers = int(env_max_workers)
 
             console.print(f"Going to download {str(count)} files to {self.local_images_path.as_posix()}.")
-            exhaust_generator(progress=progress(), count=count, multi_threaded=multi_threaded, worker_count=max_workers)
+            successes, errors = exhaust_generator(
```

Comment:
Basically a monad like structure then in the end, like it

Reply:
I'm not sure if I'd rename the function overall now, but let's leave it for a less urgent time.

Reply:
It does exhaust a generator and return values, so it is doing what it says on the box. We just happen to only ever use it to trigger downloads and the function was originally written such that the generator has to return a callable function, I haven't written any tests or checked whether the change I made to 'f(x)' will result in not requiring callable restriction because honestly, it didn't seem worth it given we only ever use it for this one purpose.